imap/mboxname.c

```diff
@@ -1499,190 +1499,121 @@ EXPORTED int mboxname_isdeletedmailbox(const char *name, time_t *timestampp)
     return res ? 1 : 0;
 }
 
-/*
- * If (internal) mailbox 'name' is a CALENDAR mailbox
- * returns boolean
- */
-EXPORTED int mboxname_iscalendarmailbox(const char *name, int mbtype)
-{
-    if (mbtype_isa(mbtype) == MBTYPE_CALENDAR) return 1;  /* Only works on backends */
-    int res = 0;
-
-    mbname_t *mbname = mbname_from_intname(name);
-    const strarray_t *boxes = mbname_boxes(mbname);
-    const char *prefix = config_getstring(IMAPOPT_CALENDARPREFIX);
-    if (strarray_size(boxes) && !strcmpsafe(prefix, strarray_nth(boxes, 0)))
-        res = 1;
-
-    mbname_free(&mbname);
-    return res;
-}
+static const enum imapopt mailbox_prefix_opts[] = {
+    IMAPOPT_NOTESMAILBOX,
+    /* non-IMAP prefixes */
+    IMAPOPT_CALENDARPREFIX,
+    IMAPOPT_ADDRESSBOOKPREFIX,
+    IMAPOPT_DAVDRIVEPREFIX,
+    IMAPOPT_DAVNOTIFICATIONSPREFIX,
+    IMAPOPT_JMAPNOTIFICATIONFOLDER,
+    IMAPOPT_JMAPSUBMISSIONFOLDER,
+    IMAPOPT_JMAPPUSHSUBSCRIPTIONFOLDER,
+    IMAPOPT_JMAPUPLOADFOLDER,
+    IMAPOPT_SIEVE_FOLDER,
+    IMAPOPT_ZERO
+};
 
-/*
- * If (internal) mailbox 'name' is a ADDRESSBOOK mailbox
- * returns boolean
- */
-EXPORTED int mboxname_isaddressbookmailbox(const char *name, int mbtype)
+static int _mbname_isa(const mbname_t *mbname, const enum imapopt prefix_opts[])
```

Comment:
This returns true if the `mbname`'s prefix matches _any of_ the specified `prefix_opts`.  That's fine, but the other way "isa" could be interpreted is that it returns true only if if matched _all of_ them.

So to make it clearer to future readers, I think I would call this `_mbname_isanyof` instead of `_mbname_isa`.  That way they don't have to stop and read the implementation to figure out which kind of "isa" is it.

Reply:
Is now ` _mbname_prefix_isanyof()`